crates/storage-query-datafusion/src/partition_state/row.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2023 - 2025 Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use crate::table_util::format_using;
+
+use super::schema::PartitionStateBuilder;
+use restate_types::{
+    GenerationalNodeId, cluster::cluster_state::PartitionProcessorStatus, identifiers::PartitionId,
+};
+
+#[inline]
+pub(crate) fn append_partition_row(
+    builder: &mut PartitionStateBuilder,
+    output: &mut String,
+    node_id: GenerationalNodeId,
+    partition_id: PartitionId,
+    state: &PartitionProcessorStatus,
+) {
+    let mut row = builder.row();
+
+    row.partition_id(partition_id.into());
+    row.plain_node_id(format_using(output, &node_id.as_plain()));
+    row.gen_node_id(format_using(output, &node_id));
+    row.target_mode(state.planned_mode.to_string().to_uppercase());
```

Comment:
I'm not sure why uppercase, are you trying to make it consistent with something else?

Reply:
I think i was influenced by another view idd but I can't remember what. I will change it back to use `impl Display`